src/components/MainContent/index.js

```diff
@@ -1,27 +1,42 @@
 import styles from "./mainContent.module.css";
 import { useState } from "react";
-import { getCoinFromSearch } from "../../utils/api";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchSingleCoin } from "../../features/singleCoinSlice";
+import Loader from "../Loader";
+import Error from "../ErrorAlert";
+
 function MainContent() {
-    const [searchedCoin, setsearhedCoin] = useState("");
+    const [searchedCoin, setSearchedCoin] = useState("");
+    const dispatch = useDispatch();
+    const { loading, data, error } = useSelector(
+        (state) => state.aSearchedCoin
+    );
+
+    const handleInputChange = (e) => {
+        setSearchedCoin(e.target.value);
+    };
 
     const handleSubmit = () => {
-        //TODO - add loading state and loader
-        getCoinFromSearch(searchedCoin);
+        dispatch(fetchSingleCoin(searchedCoin));
     };
 
     return (
-        <div className={styles.searchBarContainer}>
-            <input
-                className={styles.searchBar}
-                type="text"
-                value={searchedCoin}
-                onChange={(e) => setsearhedCoin(e.target.value)}
-                placeholder="Search a coin"
-            />
-            <button onClick={handleSubmit} className={styles.submitButton}>
-                Submit
-            </button>
-        </div>
+        <section>
```

Comment:
Noticed this now, you added a section without a class that just wraps your div.searchBarContainer

Why have a double container? Either add the class to the section and delete the div (which is what I'd do) or remove the section. Having both is unneeded semantically, and structurally.